factory/cluster.go

```diff
@@ -18,7 +19,7 @@ func ClusterConfig(g GinkgoTInterface) *Cluster {
 	once.Do(func() {
 		var err error
 		cluster, err = newCluster(g)
-		if err != nil {
+		if err != nil && customflag.ServiceFlag.ClusterConfig.Destroy {
```

Comment:
here you are saying &&  but we want to destroy here even if `customflag.ServiceFlag.ClusterConfig.Destroy` is empty

Reply:
this flag is already being used on suites - https://github.com/rancher/distros-test-framework/blob/6b77ea38bbd8b277012736a360c6a03069880595/entrypoint/certrotate/certrotate_suite_test.go#L39

Reply:
So I would like to have the option to keep the cluster in case I want to debug the failure. Should we add a separate debug flag for that then? Else right now its destroying on failures by default and I have no way to debug why it failed

Reply:
so the  "install/ initialization " failures  will be logged pretty much every time for you 

do we want on to still have the cluster up to debug further? 

if so, yeah we should leave as you did with `&&` 

when i add that i was thinking that we wanted always when errors come from the install/ init tf and etc stuff, we wanted to only log and kill

Reply:
if we go that way you proposed , i would say that does not even need to add the kill cluster in here
since we are already passing kill=true , its a overhead and duplicate stuff

Reply:
if thats the case what i am saying is then go like 
```

func ClusterConfig() *Cluster {
	once.Do(func() {
		var err error
		cluster, err = newCluster()
		if err != nil {
			l.Errorf("building cluster failed!: %vn", err)
			os.Exit(1)
		}
	})

	return cluster
}

Reply:
Yeah so there are couple of types of failure we can get, these comes to my mind:

1.  Failure sure to terraform variable/value mismatch (like unsupported or missing value on tf.vars)
2. Failure on the installation process
3. Failure on the test itself

So, for 1. we can destroy the cluster as it does not make sense to proceed as the var/value is wrong
But for 2 and 3, we should have the ability to debug further to find where and why it failed even though we are logging it. Will need the cluster to share the logs as well to the devs for debugging purpose. Hence I was proposing a `debug` flag that can assist with keeping or destroying the cluster... lmk what you think?

Reply:
i thought i had answered before , sorry 

the debug option would be the optimal here, but for now i would say lets only destroy if we pass the true flag in?

Reply:
if we found error we exit always, but destroy only if flag ==true